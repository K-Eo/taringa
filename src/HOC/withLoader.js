import React, { Fragment } from 'react'

import Loader from '../common/Loader'

const withLoader = (
  getShowLoader = () => false,
  className,
  size
) => BaseComponent => {
  const LoaderHOC = props => {
    const showLoader = getShowLoader(props)

    return (
      <Fragment>
        <BaseComponent {...props} />
        {showLoader && <Loader className={className} size={size} />}
      </Fragment>
    )
  }

  LoaderHOC.defaultProps = {
    count: 0,
    items: [],
    status: 'success',
  }

  return LoaderHOC
}

export default withLoader
