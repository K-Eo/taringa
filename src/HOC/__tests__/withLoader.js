import React from 'react'

import withLoader from '../withLoader'

const BaseComponent = () => {
  return <div>BaseComponent</div>
}

describe('withLoader', () => {
  it('exists', () => {
    expect(withLoader).toBeDefined()
  })

  it('renders default', () => {
    const Component = withLoader()(BaseComponent)
    const wrapper = shallow(<Component />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders loader when is fetching', () => {
    const Component = withLoader()(BaseComponent)
    const wrapper = shallow(<Component status="fetching" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders loader when has success and has more content', () => {
    const Component = withLoader()(BaseComponent)
    const wrapper = shallow(
      <Component status="success" totalCount={10} items={[1, 2, 3, 4, 5]} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('does not render loader when has success and no more content', () => {
    const Component = withLoader()(BaseComponent)
    const wrapper = shallow(
      <Component status="success" totalCount={5} items={[1, 2, 3, 4, 5]} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('does not render loader when has failure and more content', () => {
    const Component = withLoader()(BaseComponent)
    const wrapper = shallow(
      <Component status="failure" totalCount={10} items={[1, 2, 3, 4, 5]} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
