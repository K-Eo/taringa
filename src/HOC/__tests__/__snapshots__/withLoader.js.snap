// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoader adds custom className en size to Loader 1`] = `
<React.Fragment>
  <BaseComponent
    count={0}
    items={Array []}
    status="success"
  />
  <Loader
    className="foo"
    size="bar"
  />
</React.Fragment>
`;

exports[`withLoader adds custom className en size to Loader 2`] = `
<React.Fragment>
  <BaseComponent
    count={0}
    foo={false}
    items={Array []}
    status="success"
  />
  <Loader
    className="foo"
    size="bar"
  />
</React.Fragment>
`;

exports[`withLoader renders default 1`] = `
<React.Fragment>
  <BaseComponent
    count={0}
    items={Array []}
    status="success"
  />
</React.Fragment>
`;

exports[`withLoader renders loader given getShowLoader 1`] = `
<React.Fragment>
  <BaseComponent
    count={0}
    foo={true}
    items={Array []}
    status="success"
  />
  <Loader />
</React.Fragment>
`;

exports[`withLoader renders loader given getShowLoader 2`] = `
<React.Fragment>
  <BaseComponent
    count={0}
    foo={false}
    items={Array []}
    status="success"
  />
</React.Fragment>
`;
